otherlibs/systhreads/st_pthreads.h

```diff
@@ -297,15 +315,27 @@ static int st_event_wait(st_event e)
 static void * caml_thread_tick(void * arg)
 {
   int *domain_id = (int *) arg;
+  struct timeval curtime;
+  struct timespec deadline;
 
   caml_init_domain_self(*domain_id);
   caml_domain_state *domain = Caml_state;
 
-  while(! atomic_load_acquire(&Tick_thread_stop)) {
-    st_msleep(Thread_timeout);
-
+  caml_plat_lock (&Tick_thread_control.mu);
+  while(1){
+    if (Tick_thread_control.state == Tick_stop) break;
+    gettimeofday (&curtime, NULL);
```

Comment:
The clock for `caml_plat_timedwait` is set inside `caml_plat_cond_init_aux` in a platform-dependent manner. Please use e.g. `clock_gettime` to make sure the deadline is on the same clock.

Reply:
Unfortunately, `clock_gettime` is not available on MacOS, so I had to reproduce the platform-dependent part.